core/src/commonMain/kotlin/behavior/interaction/InteractionBehavior.kt

```diff
@@ -20,14 +20,14 @@ public interface InteractionBehavior : KordEntity, Strategizable {
     public val token: String
 
     /** The id of the channel the interaction was sent from. */
-    public val channelId: Snowflake
+    public val channelId: Snowflake?
 
     /** The behavior of the channel the interaction was sent from. */
-    public val channel: MessageChannelBehavior get() = MessageChannelBehavior(channelId, kord)
+    public val channel: MessageChannelBehavior? get() = channelId?.let { MessageChannelBehavior(it, kord) }
```

Comment:
i think these should be overridden with non-nullable in subtypes that need to have a channel id (i.e. `ComponentInteractionBehavior`, `ChatInputCommandInteraction` and `MessageCommandInteraction`)

Reply:
maybe a new interface like `ChannelInteractionBehavior` would be nice for this